### PR TITLE
DBSchenker

### PR DESCRIPTION
**Pull request by r0xsh**
## DBSchenker

- [ ] Update the UI
    - [x] Transporter panel
    - [x] Imported from data
    - [x] Settings from store
    - [ ] Add `reference` field on `Delivery` in create view (if it's a transporter's store)
- [x] Import Shenker messages as CoopCycle Tasks
    - [x] Use Store openning range as delivery pickup-dropoff range
    - [x] Tag premium deliveries
- [ ] Add a `docking` state on `Task` lifecycle
- [x] Allow `Task` to exists without `Address`
- [x] Add feature DBSchenkerClient to generate REPORT (https://github.com/coopcycle/edifact-generator/commit/ab95a38b2640ba8bc150f7b8dc5c0d8412eadc6e)
- [x] Add feature DBSchenkerClient to read SCONTR (https://github.com/coopcycle/edifact-mapping/commit/ab5481641392296c0e5c66bfbef02a31533e2099)
- [x] Add feature DBShenkerClient to sync (https://github.com/coopcycle/DBSchenkerClient/commit/4e2f45e1c515aeb1e1641f15666a7f0785b2f752) 
- [ ] ~Create a scheduled task that sync with transporter's FTP ([zenstruck/schedule-bundle](https://github.com/zenstruck/schedule-bundle))~
- [x] Hook `Task` lifecycle with DBSchenkerClient
    - [ ] On state change
        - [ ] Dock -> Pending(todo)
        - [ ] Pending -> In delivery
        - [x] In delivery -> Delivered
    - [ ] On reschedule
    - [ ] On fail
    - [ ] On incident
    
### Next iteration
- [ ] Improve AddressBook.js to support malformed address fix
- [ ] Fix `Tags` on `TaskType` not showing up
- [ ] Add incident when tagged with `REVIEW_NEEDED` (failure_reasons need to be refactored)

### CODE REVIEW

**Comment by r0xsh** [comment]
After few iterations. We arrived to the conclusion that the dashboard will may look like the following
dbsh

**Comment by r0xsh** [comment]
image

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,6 +51,8 @@ jobs:
       - uses: actions/checkout@v2
       - name: Create .env file
         run: cp .env.dist .env
+      - name: Add secret for composer
+        run: composer config github-oauth.github.com ${{ secrets.GH_TOKEN_DBSHENKER }}
```
Comment (alexsegura) [inline .github/workflows/test.yml:55]: Can be removed now that the repo is public

File: app/DoctrineMigrations/Version20231115200732.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE store ADD db_shenker BOOLEAN DEFAULT \'false\' NOT NULL');
```
Comment (alexsegura) [inline app/DoctrineMigrations/Version20231115200732.php:23]: Should be `db_schenker`. And it seems not used.

**Review by alexsegura — APPROVED** [review]

**Pull request merged**